### PR TITLE
Add Bitfighter

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/bitfighter/bitfighter/pull/667
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/bitfighter/bitfighter/pull/667 https://github.com/bitfighter/bitfighter/pull/668

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build org.bitfighter.Bitfighter

File: org.bitfighter.Bitfighter.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    cleanup:
+      - '*.a'
+      - '*.la'
+      - /include
+      - /lib/pkgconfig
+
+  - name: mpg123
+    sources:
+      - type: archive
+        url: https://sourceforge.net/projects/mpg123/files/mpg123/1.26.5/mpg123-1.26.5.tar.bz2
```
Comment (hfiguiere) [inline org.bitfighter.Bitfighter.yaml:30]: this is in the runtime

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/org.bitfighter.Bitfighter.appdata.xml: FAILED:
• style-invalid         : <caption> is too short [Core];shortest allowed is 5 chars
Validation of files failed
```

**Comment by hfiguiere** [comment]
as for the aarch64 failure, it seems that this software is very portable
```
[ 19%] Building C object tomcrypt/CMakeFiles/tomcrypt.dir/src/pk/asn1/der/bit/der_encode_bit_string.c.o
lj_arch.h:55:2: error: #error "No support for this architecture (yet)"
   55 | #error "No support for this architecture (yet)"
      |  ^~~~~
lj_arch.h:294:2: error: #error "No target architecture defined"
  294 | #error "No target architecture defined"
      |  ^~~~~
Makefile:254: *** Unsupported target architecture.  Stop.
```

**Comment by Mailaender** [comment]
Building with https://github.com/libtom/libtomcrypt/releases/tag/v1.18.2 in the hopes that it has more ARM support than the bundled library.

**Comment by Mailaender** [comment]
bot, build org.bitfighter.Bitfighter

**Comment by Mailaender** [comment]
Updating https://github.com/LuaJIT/LuaJIT/commits/v2.1 in hopes of ARM support.

**Comment by Mailaender** [comment]
bot, build org.bitfighter.Bitfighter

**Comment by barthalion** [comment]
bot, build org.bitfighter.Bitfighter

**Pull request closed without merging**